src/librustc_codegen_ssa/back/link.rs

```diff
@@ -839,7 +839,13 @@ pub fn linker_and_flavor(sess: &Session) -> (PathBuf, LinkerFlavor) {
                             "emcc"
                         }
                     }
-                    LinkerFlavor::Gcc => "cc",
+                    LinkerFlavor::Gcc => {
+                        if cfg!(target_os = "solaris") {
```

Comment:
It would be better to have a comment here to explain why it is.

Reply:
+1-- a comment would be great.

Reply:
Indeed!  I have added a comment describing the situation.